src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -317,7 +335,8 @@ private static void ScanForExpiredItems(MemoryCache cache)
         {
             DateTimeOffset now = cache._lastExpirationScan = cache._options.Clock.UtcNow;
 
-            foreach (KeyValuePair<object, CacheEntry> item in cache._entries)
+            var entries = cache._entries; // Clear() can update the reference in the meantime
```

Comment:
Nit: `var` should be replaced with the concrete type?
(same on other places in this PR)